src/framework/constraints.ml

```diff
@@ -991,6 +991,39 @@ struct
     if D.is_bot d then raise Deadcode else d
 end
 
+module DeadBranchLifter (S: Spec): Spec =
+struct
+  include S
+
+  let name () = "DeadBranch (" ^ name () ^ ")"
+
+  module Locmap = Deadcode.Locmap
+
+  let dead_branches = function true -> Deadcode.dead_branches_then | false -> Deadcode.dead_branches_else
+
+  let locmap_modify_def d k f h =
+    if Locmap.mem h k then
+      Locmap.replace h k (f (Locmap.find h k))
+    else
+      Locmap.add h k d
+
+  let branch ctx exp tv =
+    if !GU.in_verifying_stage && get_bool "dbg.print_dead_code" then (
+      Locmap.replace Deadcode.dead_branches_cond !Tracing.current_loc exp;
```

Comment:
I know this PR didn't change it, but why is this called `dead_branches_cond`? Does it not somehow collect the conditions of all branches?

Reply:
Yes, because the dead branch printing in `Control` is done purely by location so it wouldn't know what conditions to print otherwise.